Common/Data/DataQueueHandlerSubscriptionManager.cs

```diff
@@ -0,0 +1,150 @@
+                int count;
+                if (SubscribersByChannel.TryGetValue(channel, out count))
+                {
+                    if (count > 1)
+                    {
+                        SubscribersByChannel.TryUpdate(channel, count - 1, count);
+                        return;
+                    }
+
+                    if (Unsubscribe(new[] { dataConfig.Symbol }, dataConfig.TickType))
+                    {
+                        SubscribersByChannel.TryRemove(channel, out count);
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                Log.Error(exception);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Describes the way <see cref="IDataQueueHandler"/> implements subscription
+        /// </summary>
+        /// <param name="symbols">Symbols to subscribe</param>
+        /// <param name="tickType">Type of tick data</param>
+        /// <returns>Returns true if subsribed; otherwise false</returns>
+        protected abstract bool Subscribe(IEnumerable<Symbol> symbols, TickType tickType);
+
+        /// <summary>
+        /// Describes the way <see cref="IDataQueueHandler"/> implements unsubscription
+        /// </summary>
+        /// <param name="symbols">Symbols to unsubscribe</param>
+        /// <param name="tickType">Type of tick data</param>
+        /// <returns>Returns true if unsubsribed; otherwise false</returns>
+        protected abstract bool Unsubscribe(IEnumerable<Symbol> symbols, TickType tickType);
+
+        /// <summary>
+        /// Brokerage maps <see cref="TickType"/> to real socket/api channel
+        /// </summary>
+        /// <param name="tickType">Type of tick data</param>
+        /// <returns></returns>
+        protected abstract string ChannelNameFromTickType(TickType tickType);
+
+        /// <summary>
+        /// Checks if there is existing subscriber for current channel
+        /// </summary>
+        /// <param name="symbol">Symbol</param>
+        /// <param name="tickType">Type of tick data</param>
+        /// <returns>return true if there is one subscriber at least; otherwise false</returns>
+        public bool IsSubscribed(Symbol symbol, TickType tickType)
+        {
+            return SubscribersByChannel.ContainsKey(GetChannel(
+                symbol,
+                tickType));
+        }
+
+        private Channel GetChannel(SubscriptionDataConfig dataConfig) => GetChannel(dataConfig.Symbol, dataConfig.TickType);
+
+        private Channel GetChannel(Symbol symbol, TickType tickType)
+        {
+            return new Channel(
+                ChannelNameFromTickType(tickType),
+                symbol);
+        }
+
+        /// <summary>
+        /// Returns subscribed symbols
+        /// </summary>
+        /// <returns>list of <see cref="Symbol"/> currently subscribed</returns>
+        public IEnumerable<Symbol> GetSubscribedSymbols()
```

Comment:
nit - `public` methods should be before `private/protected` ones, same for `IsSubscribed`